podpac/core/node.py

```diff
@@ -281,6 +281,10 @@ def eval(self, coordinates, **kwargs):
         key = "output"
         cache_coordinates = coordinates.transpose(*sorted(coordinates.dims))  # order agnostic caching
 
+        if isinstance(self, (podpac.data.DataSource, podpac.core.compositor.BaseCompositor)) and self.dims:
+            extra_dims = [dim for dim in cache_coordinates.dims if dim not in self.dims]
+            cache_coordinates = cache_coordinates.drop(extra_dims)
+
```

Comment:
@mpu-creare You'll notice that I did both datasources and compositors. Let me know if you think I should only do it for datasources and not compositors.